core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -816,6 +816,34 @@ public void changeCacheFlag(int columnIndex, boolean cache) {
         updateMetaStructureVersion();
     }
 
+    @Override
+    public void convertTable(int walFlag) {
```

Comment:
Now thinking of this method in `TableWriter` can be a problem.
If table is WAL and it's suspended because of problems with WAL files or Sequencer files the only choice the user has is to convert it to non-WAL. But it won't work when table is suspended.

I'd move this code to SqlCompiler and execute it while compiling sql commands, similar what we do in `alterSystemLockWriter`

Reply:
fixed in [927dc44](https://github.com/questdb/questdb/pull/2947/commits/927dc445c6486e26a759d9a70cfccc25135aa03e)